src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/LambdaCapturedVariable.cs

```diff
@@ -65,6 +65,11 @@ private static string GetCapturedVariableFieldName(Symbol variable, ref int uniq
                 {
                     return GeneratedNames.MakeHoistedLocalFieldName(local.SynthesizedKind, uniqueId++);
                 }
+
+                if (local.SynthesizedKind == SynthesizedLocalKind.InstrumentationPayload)
+                {
+                    return GeneratedNames.MakeSynthesizedInstrumentationPayloadLocalName(uniqueId++);
+                }
```

Comment:
I thought that the payload is now not inside instrumented method but in PrivateImplementationDetail class?